src/main/scala/com.snowplowanalytics.sauna/responders/hipchat/SendRoomNotificationResponder.scala

```diff
@@ -0,0 +1,238 @@
+   *
+   * @param roomNotification A room notification represented as a case class.
+   * @param source           The observer event that triggered this responder event.
+   */
+  case class RoomNotificationReceived(
+    roomNotification: RoomNotification,
+    source: ObserverBatchEvent) extends ResponderEvent[ObserverBatchEvent]
+
+  /**
+   * A responder result denoting that a HipChat room notification was successfully sent
+   * by the responder.
+   *
+   * @param source  The responder event that triggered this.
+   * @param message A success message.
+   */
+  case class RoomNotificationSent(
+    source: RoomNotificationReceived,
+    message: String) extends ResponderResult
+
+  /**
+   * A self-describing JSON structure, containing a reference to a JSON
+   * schema and data that should validate against the given schema.
+   *
+   * @param schema A self-describing schema key.
+   * @param data   Data that should comply to the schema.
+   */
+  case class SelfDescribing(
+    schema: SchemaKey,
+    data: JsValue)
+
+  implicit val selfDescribingReads: Reads[SelfDescribing] = (
+    (JsPath \ "schema").read[String].map[SchemaKey](schemaKey => SchemaKey.fromUri(schemaKey).get) and
+      (JsPath \ "data").read[JsValue]
+    ) (SelfDescribing.apply _)
+
+  /**
+   * A Sauna command.
+   *
+   * @param envelope Identifies the command and provides instructions on how to execute it.
+   * @param command  Specifies the exact action to execute.
+   */
+  case class SaunaCommand(
+    envelope: SelfDescribing,
+    command: SelfDescribing)
+
+  implicit val saunaCommandReads: Reads[SaunaCommand] = (
+    (JsPath \ "envelope").read[SelfDescribing] and
+      (JsPath \ "command").read[SelfDescribing]
+    ) (SaunaCommand.apply _)
+
+  /**
+   * TODO: define what this is
+   */
+  object Semantics extends Enumeration {
+    type Semantics = Value
+    val AT_LEAST_ONCE = Value("AT_LEAST_ONCE")
+  }
+
+  implicit val semanticsReads: Reads[Semantics] = Reads.enumNameReads(Semantics)
+
+  /**
+   * TODO: define what this is
+   *
+   * @param semantics  TODO: define what this is
+   * @param timeToLive The command's lifetime (in ms): if a command is
+   *                   received after `timeToLive` ms from its' creation have
+   *                   passed, it will be discarded. (If None, a command won't be
+   *                   discarded based on creation time.)
+   */
+  case class ExecutionParams(
+    semantics: Semantics,
+    timeToLive: Option[Int]
+  )
+
+  implicit val timeToLiveReads: Reads[Option[Int]] = JsPath.readNullable[Int]
+  implicit val executionParamsReads: Reads[ExecutionParams] = Json.reads[ExecutionParams]
+
+  /**
+   * A Sauna command envelope.
+   *
+   * @param commandId   A unique identifier (uuid4) for the command.
+   * @param whenCreated The command's creation date.
+   * @param execution   TODO: define what this is
+   * @param tags        TODO: define what this is
+   */
+  case class CommandEnvelope(
```

Comment:
Yeah, I like it! Maybe even just `c.s.sauna.responders.Command`, as `Responder` suffix is signaling about some concrete responder and these are common entities.